### DISCUSSION TITLE
Danfoss Ally TRVs not heating as intended

### DISCUSSION THREAD

**Discussion by Tul47** [Koenkk/zigbee2mqtt#19495] [Devices]
I've been testing a few Danfoss Ally TRVs on radiators using external sensor - No problems sending external temps to the TRVs and they receive them... But they don't really seem to act on them.  Let me show a screenshot:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/139585663/6f584723-59a6-4bfc-b856-9825dd1e570c)

I've tried to send the external temperature with and without x100, it doesn't really seem to matter.  It never really opens up the valve to get to the setpoint (calculates a low heating demand), it's clearly still looking at it's internal temperature measurement...
Any ideas?

**Comment by lux73** [comment]
have you set the "**Radiator covered**" switch before?

have you read the **whole Information is described** the correct procedure to use an external Sensor:

The temperature sensor of the TRV is — due to its design — relatively close to the heat source (i.e. the hot water in the radiator). Thus there are situations where the `local_temperature` measured by the TRV is not accurate enough: If the radiator is covered behind curtains or furniture, if the room is rather big, or if the radiator itself is big and the flow temperature is high, then the temperature in the room may easily diverge from the `local_temperature` measured by the TRV by 5°C to 8°C. In this case you might choose to use an external room sensor and send the measured value of the external room sensor to the `External_measured_room_sensor` property.The way the TRV operates on the `External_measured_room_sensor` depends on the setting of the `Radiator_covered` property: If `Radiator_covered` is `false` (Auto Offset Mode): You *must* set the `External_measured_room_sensor` property *at least* every 3 hours. After 3 hours the TRV disables this function and resets the value of the `External_measured_room_sensor` property to -8000 (disabled). You *should* set the `External_measured_room_sensor` property *at most* every 30 minutes or every 0.1K change in measured room temperature.If `Radiator_covered` is `true` (Room Sensor Mode): You *must* set the `External_measured_room_sensor` property *at least* every 30 minutes. After 35 minutes the TRV disables this function and resets the value of the `External_measured_room_sensor` property to -8000 (disabled). You *should* set the `External_measured_room_sensor` property *at most* every 5 minutes or every 0.1K change in measured room temperature.

**If your external Sensor is not sending properly Values within the required Timeframe your Setup will not work** ^^

**Reply by Tul47** [reply]
Hi Michael,
Yes, radiator covered is set - Should be visible in the screenshot above.
I've read all I could about it and I'm sending the values minimum every 15min, and they seem to be received properly - It's not resetting to -8000, and I can see the values changing in the screen shown...
In the screenshot it's at 2144, as I believe it should be multiplied by 100 - But it doesn't seem to care about it.

**Comment by hotchkj** [comment]
You're not alone in this. There are also other related [discussions](https://github.com/Koenkk/zigbee2mqtt/discussions/14794). I have a whole set of the newer iteration of the Allys, and they're all pretty random in what they decide to do. For what it's worth, I sent a message to Danfoss Support about this and got a fairly meaningless reply; given the lack of further firmware updates I suspect the Allys are abandoned internally. They certainly don't do the job as advertised by their own [spec](https://assets.danfoss.com/documents/284715/AM375549618098en-000103.pdf).

One thing to consider is the guidance that Zigbee2MQTT has was written back when the original non-G Allys came out and were updated with later firmware `1.20`. The new iteration have theoretically identical firmware (stated by Danfoss Support, I'll provide a quote below), but I can find no evidence of them responding to `external_measured_room_sensor` changing, without some other input changing as well.

Logically, the Ally Gateway, Room Sensor, and TRV must all communicate and set temperatures in order to do what the Ally integrated system proposes, yet I can't see how this actually would happen in practice. Consider a sensor and a TRV, the internal temp is 20, the setpoint is 22, the external temp set is 24 - logically the TRV should have closed but was stuck at 93% Heating Demand.

The moment I cycled `radiator_covered` (`True` -> `False` -> `True`) and changed the setpoint to `21.5` (essentially giving it some further input), suddenly the TRV woke up and dropped to 5% Heating Demand which it should have done hours ago. `algorithm_scale_factor` is set to `1` so it's supposed to be a fast response. I notice this has additional documentation which implies there's a `quick_open` feature (`Range 1-10 (lower 4 bit allocated to scale factor) Bit4=Quick open feature disable. 1=disable. 0=enable)`) but I am assuming when set to `1` this is indeed enabled.

So I don't know what to make of them 🤷 I can't say I'm inclined to re-write automations to 'waggle' the setpoint every so often just to get them to respond properly.


When I asked about the odd firmware numbers on the Gs:

>Ally eTRVs FW version 1,20 is the same as Ally-G eTRV 0,20. Different numbers due to a bit different HW modification but functionality and scope are the same. 

> OTA file is only for Ally eTRVs FW version 1,20. For Ally-G eTRV 0,20 it is still needed to do some manual change in OTA file following pdf instruction together with OTA file.  

> you need to make the changes on your own - So manual change the image type in the firmware from 0x100 -> 0x120

**Reply by hotchkj** [reply]
I suppose as well that there is another possibility to the above - which is that the TRV _has_ responded, but is failing to actually _advertise_ changes i.e. all the numbers we're seeing are not actually true until it refreshes some internal state, or something in the TRV->Zigbee-> MQTT layers reports old data and then suddenly updates, which would make inferring what it is doing pretty challenging.
